docs/tutorials/getting-started-with-flask.rst

```diff
@@ -0,0 +1,454 @@
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker]
+    :end-before: [docs:stop-docker-end]
+    :dedent: 2
+
+Chisel the rock
+===============
+
+This is an optional but recommended step, especially if you're looking to
+deploy your rock into a production environment. With :ref:`chisel_explanation`
+you can produce lean and production-ready rocks by getting rid of all the
+contents that are not needed for your Flask application to run. This results
+in a much smaller rock with a reduced attack surface.
+
+The first step towards chiselling your rock is to ensure you are using a
+``bare`` base.
+In ``rockcraft.yaml``, change the ``base`` to ``bare`` and add
+``build-base: ubuntu@22.04``:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:change-base]
+    :end-before: [docs:change-base-end]
+    :dedent: 2
+
+Pack the rock with the new ``bare`` base:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:chisel-pack]
+    :end-before: [docs:chisel-pack-end]
+    :dedent: 2
+
+As before, verify that the new rock was created:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:ls-bare-rock]
+    :end-before: [docs:ls-bare-rock-end]
+    :dedent: 2
+
+You'll verify that the new Flask rock is now approximately **15% smaller**
+in size! And that's just because of the simple change of ``base``.
+
+And the functionality is still the same. As before, you can confirm this by
+running the rock with Docker
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: text
+    :start-after: [docs:docker-run-chisel]
+    :end-before: [docs:docker-run-chisel-end]
+    :dedent: 2
+
+and then using the same ``curl`` request:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: text
+    :start-after: [docs:curl-flask-bare-rock]
+    :end-before: [docs:curl-flask-bare-rock-end]
+    :dedent: 2
+
+Unsurprisingly, the Flask application should still respond with
+``Hello, world!``.
+
+Cleanup
+~~~~~~~
+
+And that's it. You can now stop container and remove the corresponding image:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker-chisel]
+    :end-before: [docs:stop-docker-chisel-end]
+    :dedent: 2
+
+.. _update-flask-application:
+
+Update Flask application
+========================
+
+As a final step, let's say you want to update your application. For example,
+you want to add a new ``/time`` endpoint which returns the current time.
+
+Start by opening the ``app.py`` file in a text editor and update the code to
+look like the following:
+
+.. literalinclude:: code/getting-started-with-flask/time-app.py
+    :language: python
+
+Since you are creating a new version of your application, open the
```

Comment:
To add base and buid-base, a sed one liner is used. However to update the versions, the user is asked to do it. I think it should be the same for both cases (whatever is nicer).

Reply:
TBH I think this is ok like this. The reason is because the `sed` command is necessary because that change must be applied verbatim. All the other file editing changes are not as strict, i.e. if the user sets the version to 0.1.1, this will still work...we use 0.2. as an example. OTOH, the "base" modification **must** literally take those values.